sql-tuning-best-practice.md

```diff
@@ -0,0 +1,714 @@
+
++--------------------------+-------------+--------------+-------------------+----------------------------------------------------------------------------------------------------+
+| id                       | estRows     | task         | access object     | operator info                                                                                      |
++--------------------------+-------------+--------------+-------------------+----------------------------------------------------------------------------------------------------+
+| StreamAgg_20             | 1.00        | root         |                   | funcs:count(Column#13)->Column#11                                                                  |
+| └─TableReader_21         | 1.00        | root         |                   | data:StreamAgg_9                                                                                   |
+|   └─StreamAgg_9          | 1.00        | cop[tikv]    |                   | funcs:count(1)->Column#13                                                                          |
+|     └─Selection_19       | 250.00      | cop[tikv]    |                   | ge(trips.start_date, 2017-07-01 00:00:00.000000), le(trips.start_date, 2017-07-01 23:59:59.000000) |
+|       └─TableFullScan_18 | 10000.00    | cop[tikv]    | table:trips       | keep order:false, stats:pseudo                                                                     |
++--------------------------+-------------+--------------+-------------------+----------------------------------------------------------------------------------------------------+
+5 rows in set (0.00 sec)
+```
+
+Additional Information in EXPLAIN ANALYZE Output
+Different from EXPLAIN, EXPLAIN ANALYZE executes the corresponding SQL statement, records its runtime information, and returns the information together with the execution plan. There runtime information is crucial for debugging query execution.
+
+Description
+- actRows: Number of rows output by the operator.
+- execution info: Detailed execution information of the operator. time represents the total wall time from entering the operator to leaving the operator, including the total execution time of all sub-operators. If the operator is called many times by the parent operator then the time refers to the accumulated time. loops is the number of times the current operator is called by the parent operator.
+- memory: Memory used by the operator.
+- disk: Disk space used by the operator.
+
+Note: Some attributes and explain table columns are omitted for improved formatting
+```SQL
+tidb:db1> EXPLAIN ANALYZE SELECT SUM(pm.m_count)/COUNT(*) FROM
+    -> (SELECT COUNT(m.name) m_count
+    ->  FROM universe.moons m
+    ->  RIGHT JOIN
+    ->  (SELECT p.id, p.name
+    ->   FROM universe.planet_categories c
+    ->   JOIN universe.planets p
+    ->   ON c.id = p.category_id AND c.name = 'Jovian') pc
+    ->  ON m.planet_id = pc.id
+    ->  GROUP BY pc.name) pm;
```

Comment:
```suggestion
```sql
EXPLAIN ANALYZE
SELECT SUM(pm.m_count)/COUNT(*) FROM
(SELECT COUNT(m.name) m_count
FROM universe.moons m
RIGHT JOIN
(SELECT p.id, p.name
FROM universe.planet_categories c
JOIN universe.planets p
ON c.id = p.category_id AND c.name = 'Jovian') pc
ON m.planet_id = pc.id
GROUP BY pc.name) pm;
```

This makes the output copy-paste-able and makes it easier for a highlighting to work correctly